docs/cr_HarborCluster_spec.md

```diff
@@ -0,0 +1,203 @@
+## Harbor-Cluster CRD
+
+Below is the spec defined for the CR `harbor-cluster`:
+
+```yaml
+# harbor version to be deployed
+# this version determines the image tags of harbor service components
+# required
+version: 1.10
+
+# external URL for access Harbor registry
+# required
+publicURL: https://harbor.registry.com
+
+# password for the root admin 
+# required
+adminPasswordSecret: adminSecret
+
+# secret reference for the TLS certs
+tlsSecret: tlsSecret
+
+# certificate issuers
+certificateIssuerRef: cert_issuer
+
+paused: false
+
+priority: 30
+
+# pod instance number
+# required
+replicas: 3
+
+# source registry of images
+imageSource:
+  registry: harbor.com
+  imagePullSecrets:
+   - pSecret
+   
+# log configurations
+log:
+  level: debug
+  
+# The option to set repository read only.
+readOnly: false
+
+# set proxy
+proxy:
+  http_proxy: 10.10.20.2
+  https_proxy: 10.10.20.2
+  no_proxy: 10.123.111.10
+    components:
+    - core
+    - jobservice
+    - clair
+
+# extra configuration options for jobservices
+jobService:
+  workerCount: 10
+  replicas: 5
+
+# extra configuration options for clair scanner
+clair:
+  updateInterval: 10
+  vulnerabilitySources:
+    - ubuntu
+    - alphine
+
+# extra configuration options for trivy scanner
+trivy:
+  github_token: 123
+
+# extra configuration options for chartmeseum
+chartMuseum:
+  absoluteURL: true
+  
+# cache service(Redis) configurations
+# required
+redis:
```

Comment:
add kind for redis/database:
  ```yaml
   redis:
     kind: inCluster # external
  ```

Reply:
For external, we only need a secret reference.

Document the secret content format to users.